### PR TITLE
StratifiedRules: Boolean connectives

### LINKED ISSUE #2640: Stratified rules: Boolean connectives
Add new rule for `and` and `or`.

### PR DESCRIPTION
**Pull request by Kukovec**
- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [ ] Documentation added for any new functionality
- [ ] [Entries added to `./unreleased/`][changelog format] for any new functionality

closes #2640 

[changelog format]: https://github.com/informalsystems/apalache/blob/main/CONTRIBUTING.md#how-to-record-a-change

### CODE REVIEW

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/bool/AndStratifiedRule.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import at.forsyte.apalache.tla.lir.oper.TlaBoolOper
+import at.forsyte.apalache.tla.lir.{BoolT1, OperEx, TlaEx}
+import at.forsyte.apalache.tla.typecomp.TBuilderInstruction
+import at.forsyte.apalache.tla.types.tla
+
+/**
+ * Implements the rule for conjunction. Similar to TLC, we short-circuit A /\ B as IF A THEN B ELSE FALSE. This allows
+ * us to introduce an optimization on-the-fly for the conjunctions that were marked with a hint. In this optimization,
+ * we push the context, assume A and check satisfiability of the SMT context. If the context is unsat, we immediately
+ * return FALSE. Otherwise, we pop the context and continue.
```
Comment (konnov) [inline tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/bool/AndStratifiedRule.scala:16]:
```suggestion
 * Implements the rule for conjunction. Similar to TLC, we short-circuit A /\ B as IF A THEN B ELSE FALSE.
```

This comment is massively outdated. Please update it with a description about whether this is the default behavior or not.

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/bool/OrStratifiedRule.scala [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import at.forsyte.apalache.tla.bmcmt.stratifiedRules.{addCell, StratifiedRule}
+import at.forsyte.apalache.tla.bmcmt.stratifiedRules.{Rewriter, RewriterScope}
+import at.forsyte.apalache.tla.lir.{BoolT1, OperEx, TlaEx}
+import at.forsyte.apalache.tla.lir.oper.TlaBoolOper
+import at.forsyte.apalache.tla.typecomp.TBuilderInstruction
+import at.forsyte.apalache.tla.types.tla
+
+/**
+ * For state-level expressions, we express A \/ B as IF A THEN TRUE ELSE B. For action-level expressions, i.e.,
+ * involving primes, we do a direct translation to A \/ B. This mimics the behavior of TLC.
```
Comment (konnov) [inline tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/bool/OrStratifiedRule.scala:14]: I believe this comment is also quite outdated. Is there any code left that distinguishes between action and non-action expressions?

File: tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/BoolRewriterTest.scala [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    val x = tla.name("x", BoolT1)
+    val y = tla.name("y", BoolT1)
+    val expr = tla.and(x, y)
+
+    val binding = Binding(Map("x" -> xCell, "y" -> yCell))
+
+    val startScope = RewriterScope(arena, binding)
+
+    // Rewrite normal and
```
Comment (konnov) [inline tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/BoolRewriterTest.scala:47]:
```suggestion
    // Rewrite normal 'and'
```

File: tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/BoolRewriterTest.scala [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+    val x = tla.name("x", BoolT1)
+    val y = tla.name("y", BoolT1)
+    val expr = tla.or(x, y)
+
+    val binding = Binding(Map("x" -> xCell, "y" -> yCell))
+
+    val startScope = RewriterScope(arena, binding)
+
+    // Rewrite normal or
```
Comment (konnov) [inline tla-bmcmt/src/test/scala/at/forsyte/apalache/tla/bmcmt/stratifiedRules/BoolRewriterTest.scala:93]:
```suggestion
    // Rewrite normal 'or'
```

**Comment by Kukovec** [comment]
@konnov we should unironically remove fmt-checking for _comments_. Why is a template telling me how to whitespace my documentation?

**Pull request merged**